### PR TITLE
feat: add sass compatibility

### PR DESCRIPTION
**Pull request by bricelp**
<3

### CODE REVIEW

File: vite.config.ts [hunk 1]
```diff
@@ -15,4 +15,11 @@ export default defineConfig({
       buildDirectory: 'build/vite-server',
     }),
   ],
+  css: {
+    preprocessorOptions: {
+      scss: {
+        additionalData: `@import "app/ui/style/_global.scss";`,
```
Comment (heiso) [inline vite.config.ts:21]: we have to test it when built, we might have to put any scss/css somewhere else

File: app/ui/ScssButton/button.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import './button.scss'
+
+export type ButtonProps = (
+  | (ButtonHTMLAttributes<HTMLButtonElement> & { as?: 'button' })
+  | (LinkProps & { as: 'link' })
+) & { primary?: boolean }
+
+export function ScssButton({ primary, className, children, ...props }: ButtonProps) {
+  const buttonClassName = `button ${primary ? 'button-primary' : 'button-secondary'} ${className}`
```
Comment (heiso) [inline app/ui/ScssButton/button.tsx]: tbh i prefer css module for this kind of styling

File: app/ui/ScssButton/button.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { Link, type LinkProps } from '@remix-run/react'
+import { type ButtonHTMLAttributes } from 'react'
+
+import './button.scss'
```
Comment (heiso) [inline app/ui/ScssButton/button.tsx]:
```suggestion
import styles from './button.module.scss'
```

File: app/ui/ScssButton/button.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  | (ButtonHTMLAttributes<HTMLButtonElement> & { as?: 'button' })
+  | (LinkProps & { as: 'link' })
+) & { primary?: boolean }
+
+export function ScssButton({ primary, className, children, ...props }: ButtonProps) {
+  const buttonClassName = `button ${primary ? 'button-primary' : 'button-secondary'} ${className}`
+
+  if (props.as === 'link') {
+    return (
+      <Link className={buttonClassName} {...props}>
```
Comment (heiso) [inline app/ui/ScssButton/button.tsx:22]:
```suggestion
      <Link className={[styles.button, styles['button-primary']].join(' ')} {...props}>
```

Not sure it's the best way to add multiple classnames

Comment (bricelp) [reply app/ui/ScssButton/button.tsx:22]: classNames lib handle this

**Pull request closed without merging**